sandbox-backend/src/main/java/com/exadel/sandbox/team5/dao/TagDAO.java

```diff
@@ -1,9 +1,24 @@
 package com.exadel.sandbox.team5.dao;
 
 import com.exadel.sandbox.team5.entity.Tag;
+import com.exadel.sandbox.team5.util.PairSL;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
+
 @Repository
 public interface TagDAO extends JpaRepository<Tag, Long> {
+
+    @Query(value = """
+            SELECT new com.exadel.sandbox.team5.util.PairSL(t.name,COUNT(o.id))
+            FROM Tag t 
+                LEFT JOIN t.discounts d
+                LEFT JOIN d.tags dt
```

Comment:
Why do you want to use `JOIN` for Tag table if you already use it with `FROM`

Reply:
indeed - refactored.